### ISSUE TITLE
Feature Request: Replace playback bar with Audio Graph

### ISSUE THREAD

**Issue by AmonBune** [Rello/audioplayer#355]
It would be super-cool, if the playback bar would display the current track's audio graph. One could normally seek with clicking somewhere and/or click & dragging. The position is indicated by a color or a fine line.
Something like this:
![image](https://user-images.githubusercontent.com/15927695/47204813-87a9dc00-d384-11e8-9b07-236002d621e7.png)

I know this request is pretty demanding, but I've read a bit about the topic and it seems that there exist libraries to create such a graph, and it should technically be possible with JS. But the question is, if it would be compatible with the current audio playback system.
If someone has some spare time to implement this, it would be pretty nice. If not, I could try to do it by myself, but I'm not experienced in developing NC apps.

What do you think about the idea?

Library for creating audiographs (wavesurfer):
https://github.com/katspaugh/wavesurfer.js#readme

SO Answers with more libraries for creating audiographs:
https://stackoverflow.com/questions/11677246/drawing-zoomable-audio-waveform-timeline-in-javascript#11692086

**Comment by Rello** [comment]
Hello,
I actually like this idea - but I see limited time at the moment.

if you want to have a look: start looking at the item
`<div class="sm2-progress-track">`
this is where you would need to set the backgroud fi you want to keep the standard soundmanager-bar-ui
if you would find a way to replace its background by the graph, it would be the first start I guess.

As said: I cant implement much here at the moment. but I am happy to support if you have any questions

**Comment by AmonBune** [comment]
I will take a look at it. 👍

**Comment by AmonBune** [comment]
I think I've found the files I have to change:

- part.sm2-bar.php
- bar-ui.js
- bar-ui.css

@Rello 
But I need a little help. I don't know how to build the app, respectively, how to run it on a nextcloud instance. What steps do you do here?
I can see that you have minified CSS files and other meta file stuff in the source code, so the whole thing has to be transpiled, right?

**Comment by Rello** [comment]
hey, easy as all files are part of the package anyway.

within index.php, remove the -min and the full file is used
https://github.com/Rello/audioplayer/blob/master/templates/index.php#L17

I think as a start you can include your js in the clickhandler here
https://github.com/Rello/audioplayer/blob/master/js/app.js#L429

via `activeLi.data('path'`) you should get the absolute path to the file. how this can be parsed and analysed in your js library? I hope you will find out...

**Comment by Rello** [comment]
Hello,
did you do some prototyping here? can I assist with something?

**Comment by AmonBune** [comment]
Hey,
Sadly I didn't have time to look into it, I had a lot of other things to do...

**Comment by AppleFire** [comment]
I would like to encourage the development of a position bar at least, if not audio graph. It's important usability that the user can start and stop audio files, and maybe drag/click the cursor back a little bit, or a lot, and replay the audio file. Currently, if sharing a folder or audio file, this is not possible. When audio files are shared with a link in Nextcloud, the files just play. If the audio is stopped, the file must start playing from the beginning. This is discouraging for the user. I am a song writer and would like to present my songs to talent executives by sharing. To me, this feature is mandatory. I wish I knew how to program.

**Comment by AmonBune** [comment]
@Rello I want to share my progress with you. I have created a demo on [https://amonbune.github.io/Audiograph-Demo/audiograph_demo.html](https://amonbune.github.io/Audiograph-Demo/audiograph_demo.html)
![image](https://user-images.githubusercontent.com/15927695/67988537-96320100-fc30-11e9-8f2e-3f99966d4ecf.png)


I'm not sure how we should integrate it into the Audioplayer app, since wavesurfer brings its own audio playback library, but I'm confident that it can be solved.

**Comment by Rello** [comment]
Hello,
AP was redesigned and the old bar-ui was removed and I switched to a much simpler progress bar + <audio> for the playback.

In case you want to look at it, it should now be much easier.
If you want to look how to integrate it, is should be "just" this function, I guess
https://github.com/Rello/audioplayer/blob/master/js/player.js#L210

let me now if you want to process this.
I don´t have time at the moment for this request unfortunately
